crates/wasmtime/src/externals.rs

```diff
@@ -283,16 +283,22 @@ impl Global {
                 ValType::I64 => Val::from(*definition.as_i64()),
                 ValType::F32 => Val::F32(*definition.as_u32()),
                 ValType::F64 => Val::F64(*definition.as_u64()),
-                ValType::ExternRef => Val::ExternRef(
-                    definition
-                        .as_externref()
-                        .clone()
-                        .map(|inner| ExternRef { inner }),
-                ),
-                ValType::FuncRef => {
-                    Val::FuncRef(Func::from_raw(store, definition.as_anyfunc() as usize))
+                ValType::Ref(rt) => {
+                    match rt.heap_type {
+                        HeapType::Extern => Val::ExternRef(
+                            definition
+                                .as_externref()
+                                .clone()
+                                .map(|inner| ExternRef { inner }),
+                        ),
+                        HeapType::Func => {
+                            Val::FuncRef(Func::from_raw(store, definition.as_anyfunc() as usize))
+                        }
+                        _ => todo!("Implement HeapType::Bot/Index for get") // TODO(dhil) fixme
```

Comment:
For this I think it's fine to generate a `Func` for a typed function index. There's no real way to propagate the type runtime type information into the Rust type system so I think it's best to return `Func` here and then embedders can used `.typed()` as today to get the typed view.